MobileCenter/MobileCenterTests/MSLogManagerDefaultTests.m

```diff
@@ -71,4 +71,30 @@ - (void)testProcessingLogWillTriggerOnProcessingCall {
   OCMVerify([mockDelegate onEnqueuingLog:log withInternalId:OCMOCK_ANY andPriority:priority]);
 }
 
+- (void)testDelegatesConcurrentAccess {
+  
+  // If
+  MSPriority priority = MSPriorityDefault;
+  NSString *groupID = @"MobileCenter";
+  MSLogManagerDefault *sut = [[MSLogManagerDefault alloc] initWithSender:OCMProtocolMock(@protocol(MSSender))
+                                                                 storage:OCMProtocolMock(@protocol(MSStorage))];
+  MSAbstractLog *log = [MSAbstractLog new];
+  for (int j = 0; j < 10; j++) {
+    id mockDelegate = OCMProtocolMock(@protocol(MSLogManagerDelegate));
+    [sut addDelegate: mockDelegate];
+  }
+  
+  // When
+  void (^block)() = ^{
+    for (int i = 0; i < 10; i++) {
```

Comment:
I think this is arbitrary numbers just to reproduce the race condition, looks like 10 log enqueuing is enough.